backend/gke/find-resources.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# Requires:
+# - gcloud credentials present
+# - jq
+
+# OPTIONS:
+# GKE_CRED_JSON gcloud credentials
+# GKE_PROJECT   gcloud project
+# OWNER         The owner of the clusters to filter on
+# OUTPUT_FILE   File to output to; outputs to standard out if not set.
+
+. ./defaults.sh
+. ../../include/common.sh
+# Do not require .envrc
```

Comment:
I would require .envrc nevertheless. The `buildfolder/.config/gcloud/* folder` is always setup by https://github.com/SUSE/catapult/blob/master/include/buildir.sh#L37, which means that we can use it, and people always expect catapult to not mess their local settings and installations.

Reply:
As far as I know, we don't include `buildir.sh`? In fact, for this there's no reason to have a builddir at all (because we don't need any of the state).

Reply:
`buildir.sh` gets run on `buildfoo` folder creation, when creating the cluster. That may mean, if `DOWNLOAD_CATAPULT_DEPS == true` (the default now), that gcloud gets downloaded and configured inside of the build folder.

Every target until now operates inside `buildfoo`, sourcing `.envrc`. The idea of this is to have a reproducible deployment and testing environment between machines & CI. And it also helps in not wiping out logins our touching config things out of catapult buildfolders.

This is useful when manually QAing things, as one may have 3-5 buildfolder clusters in parallel, with different providers, or same provider and different project because of quota issues.

Reply:
Yes, but the find & delete resources targets don't create new clusters (and in fact it wouldn't make sense to: the point is to remove stale clusters that catapult / terraform lost track of).

It is perfectly reasonable to want to run these targets locally, to look for clusters that may be orphaned; this is why the target by default just prints out the things it will delete.  It's only by opting in to actually deleting things that it can be dangerous.

Reply:
But even if you aren't deleting clusters, you will have revoked and re-logged in with other creds outside of catapult:
https://github.com/SUSE/catapult/pull/313/files#diff-cb76b3114cb1673b6ae4c55c3d7444775c8bd4ca1e72ef80cec7ddc36bf4cabdR19-R24

Also, one idea for the future is to converge all the headers
```
. ./defaults.sh
. ../../include/common.sh
. .envrc
```
from all scripts in `modules/*`, `backend/*` into [include/common.sh](https://github.com/SUSE/catapult/blob/master/include/common.sh), which is the last point in the execution flow before each distinct target.

That way all testing/operations scripts always execute against an already setup test environment, known to be good. This helps further rebases for a cleaned up catapult v2.

Reply:
Okay, I think I will agree to loading the `.envrc`, mainly because we still need a context (a build dir) in which to dump tools we need to download.  While we currently tie this to a cluster, we don't need to actually _build_ the cluster to run the new targets.  And your point about revoking the login is very important too (I've just been ignoring that while testing this, but in retrospect that wasn't the greatest).

Reply:
This should be resolved in fded88a.

Reply:
Thanks! didn't think about "tying the target to an unexistent cluster", that's a good point. But I think the trade-off pays off for now.

Approving.